### DISCUSSION TITLE
Use own registry at woodpecker-ci.org as default instead of docker-hub

### DISCUSSION THREAD

**Discussion by 6543** [woodpecker-ci/woodpecker#2144] [Polls]
https://www.docker.com/blog/we-apologize-we-did-a-terrible-job-announcing-the-end-of-docker-free-teams/  
https://blog.alexellis.io/docker-is-deleting-open-source-images/  
https://floss.social/@WoodpeckerCI/110037675573321221

**Comment by 6543** [comment]
Steps to do:
1. migrate from organization into Single Account on Docker Hub
2. Push to alternative registries
3. make alternative registries default one

**Comment by 6543** [comment]
namespace calmed at https://quay.io/organization/woodpeckerci

**Comment by jolheiser** [comment]
Have we tried applying for the open source program? I agree with moving away, but just wondering if there's a good way to mitigate issues in the interim.

**Comment by 6543** [comment]
yes last year already as I'd also would like to have seen the lable "official" but they never agreed so ... we are an open source project, but not by there definition :laughing:

**Comment by 6543** [comment]
I also like the idea of having a own registry directly serving from our woodpecker-ci.org domain ... but that needs good planing ... and we don't have any idea how much traffic that will be

**Comment by mscherer** [comment]
Something that might be helpful is https://httptoolkit.com/blog/docker-image-registry-facade/ 

(even if, as I found it, it break with podman).

**Comment by eriol** [comment]
@6543 if in the step 2 you mean to use more than one registry maybe pushing images also on codeberg could be a way to go.

**Comment by mscherer** [comment]
@eriol But codeberg is funded by volunteers, I do not think that would be fair to go there without knowing if 1) they can absord 2) how much the project would use. 

Having a local redirection server can help getting educated guess for point 2, thus helping point 1. 
If there is 100 hits per day on a image, we can estimate how much bandwidth is used (at least a upper bound). 

I have also seen some [projects using Scarf](https://about.scarf.sh/), a SaaS that provides stats, and maybe network ones (in which case, running it for 2 or 3 months could give a more precise estimate to be able to decide on self hosting and/or move ). It might be worth asking to project using it how was their experience. 

Obviously, that's trading one SaaS (Docker) for another one (Scarf), hence why I think it could be a temporary setup to have data to decide.

**Comment by lafriks** [comment]
> yes last year already as I'd also would like to have seen the lable "official" but they never agreed so ... we are an open source project, but not by there definition :laughing: 

Maybe could try again?

**Comment by mscherer** [comment]
So docker changed plan: https://www.docker.com/blog/no-longer-sunsetting-the-free-team-plan/ 

I guess that make it less urgent to move out, and let time to explore @6543  idea of having a registry.

**Comment by 6543** [comment]
![image](https://user-images.githubusercontent.com/24977596/228361163-c805816f-4b59-43d9-9450-0371f3f2a47d.png)

but I still think we should go down the route to host our images on our own as defautl images and only push them to other registrys second?

**Comment by qwerty287** [comment]
We now push our images to both dockerhub and quay.io, do we still think this is necessary?

**Comment by qwerty287** [comment]
Closing this now as it's not required currently. If we have issues with quay/dockerhub later we can consider this again.
